### ISSUE TITLE
BatchNorm: Before or after activation?

### ISSUE THREAD

**Issue by jrosebr1** [cvjena/cnn-models#3]
I see you are placing your BatchNorm *before* the activation layer. While ResNet does this, [it's been demonstrated](https://github.com/ducha-aiki/caffenet-benchmark/blob/master/batchnorm.md#bn----before-or-after-relu) that placing the BN *after* the non-linearity actually increases prediction accuracy. [Other deep learning experts](https://github.com/fchollet/keras/issues/1802#issuecomment-187966878) recommend this as well.

I'm just curious why for AlexNet and VGGNet the BN came before the non-linearity? Was this an implementation choice and determined by experiments?

**Comment by MarcelSimon** [comment]
Thanks for the remark! So far I followed the original proposal to put the batch norm before the activation layer. I feel it's best to follow common practice for the purpose of these pre-trained models. However, I've also seen the benchmarks you are referring to (which by the was was a very helpful source in many aspects). While it might be beneficial to place the batch norm after the activation layer, I did not experimented with it yet. If you want to run some experiments using the models here, feel free to add some results!

**Comment by jrosebr1** [comment]
Thanks for the clarification. And great job gathering the results!

> I feel it's best to follow common practice for the purpose of these pre-trained models.

While the original paper may suggest putting BN before the activation, I don't think this is common practice any longer. I'll try to contribute the results with the BN after the activation.

**Comment by MarcelSimon** [comment]
Cool thanks!

**Comment by zimenglan-sysu-512** [comment]
hi @MarcelSimon why the mult_lr in BN param set to be zero in train.prototxt? can you release the original train.prototxt? thanks.

**Comment by MarcelSimon** [comment]
Please check the examples of caffe for the usage of batch norm. mult_lr is always 0 for these. The provided prototxt's are the original ones.
